app/src/main/java/com/example/habithelper/adapters/BadgesAdapter.java

```diff
@@ -0,0 +1,121 @@
+package com.example.habithelper.adapters;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.drawable.Drawable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.content.res.AppCompatResources;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.habithelper.R;
+import com.example.habithelper.models.Habit;
+
+import java.util.List;
+
+public class BadgesAdapter extends RecyclerView.Adapter<BadgesAdapter.ViewHolder> {
+
+    private Context context;
+    private List<String> badgeList;
+
+    public BadgesAdapter(Context context, List<String> badgeList) {
+        this.context = context;
+        this.badgeList = badgeList;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(context).inflate(R.layout.item_badge, parent, false);
+        return new BadgesAdapter.ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        String badgeImage = badgeList.get(position);
+        holder.bind(badgeImage);
+    }
+
+    @Override
+    public int getItemCount() {
+        return badgeList.size();
+    }
+
+    public void clear(){
+        badgeList.clear();
+        notifyDataSetChanged();
+    }
+
+    public void addAll(List<String> stringList) {
+        System.out.println(stringList.size());
+        for (String s : stringList){
+            System.out.println(s);
+            badgeList.add(s);
+        }
+        System.out.println(badgeList.size());
+        System.out.println("addedAll");
+        notifyDataSetChanged();
+    }
+
+    public class ViewHolder extends RecyclerView.ViewHolder {
+        private TextView tvBadge;
+        private TextView tvBadgeDescription;
+        private ImageView ivBadge;
+
+        public ViewHolder(@NonNull View view) {
+            super(view);
+            tvBadge = view.findViewById(R.id.tvBadge);
+            tvBadgeDescription = view.findViewById(R.id.tvBadgeDescription);
+            ivBadge = view.findViewById(R.id.ivBadge);
+        }
+
+        public void bind(String badgeImage) {
+            setText(badgeImage);
+            Resources resources = context.getResources();
+            int resId = resources.getIdentifier(badgeImage, "drawable", "com.example.habithelper");
+            ivBadge.setBackgroundResource(resId);
+        }
+
+        private void setText(String badgeImage) {
+            if (badgeImage.equals("badge_magnificent_mood")) {
```

Comment:
You could have a dictionary where the badgeImage is the Key of the dictionary and the value could holds a tuple or class for the Text and description:
`
public class BadgeDetails{ 
  public String text; 
  public String description; 
  public BadgeDetails(String text, String description) { 
    this.description = description; 
    this.description = text; 
  } 
} 

Map<String, String> badgeDetailsMap = new HashMap<String, BadgeDetails>();
 tvBadge.setText(badgeDetailsMap.get(badgeImage).text);
 vBadgeDescription.setText(badgeDetailsMap.get(badgeImage).description);`
 
 and just fill the dictionary as a constant variable

Reply:
I'll put this as a stretch goal - I haven't taken data structures yet, so I'm unfamiliar with HashMaps, but I would love to learn